File: core/src/main/java/com/andiag/core/BaseRestClient.java
package com.andiag.core;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

public abstract class BaseRestClient {

    protected static OkHttpClient.Builder getBaseOkHttpClientBuilder(String loggingLevel) {
        return getBaseOkHttpClientBuilder(HttpLoggingInterceptor.Level.valueOf(loggingLevel));
    }

    protected static OkHttpClient.Builder getBaseOkHttpClientBuilder(
            HttpLoggingInterceptor.Level level) {

        return new OkHttpClient.Builder()
                .connectTimeout(12, TimeUnit.SECONDS)
                .readTimeout(12, TimeUnit.SECONDS)
                .retryOnConnectionFailure(true)
                .addInterceptor(getLoggingLevel(level));
    }

    private static HttpLoggingInterceptor getLoggingLevel(
            HttpLoggingInterceptor.Level loggingLevel) {

        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.setLevel(loggingLevel);
        return httpLoggingInterceptor;
    }

}

File: core/src/main/java/com/andiag/core/network/BaseRestClient.kt
package com.andiag.core.network

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import java.util.concurrent.TimeUnit

abstract class BaseRestClient {
    companion object {

        protected fun getBaseOkHttpClientBuilder(loggingLevel: String): OkHttpClient.Builder {
            return getBaseOkHttpClientBuilder(HttpLoggingInterceptor.Level.valueOf(loggingLevel))
        }

        fun getBaseOkHttpClientBuilder(level: HttpLoggingInterceptor.Level): OkHttpClient.Builder {
            return OkHttpClient.Builder()
                    .connectTimeout(12, TimeUnit.SECONDS)
                    .readTimeout(12, TimeUnit.SECONDS)
                    .retryOnConnectionFailure(true)
                    .addInterceptor(getLoggingLevel(level))
        }

        private fun getLoggingLevel(loggingLevel: HttpLoggingInterceptor.Level): HttpLoggingInterceptor {
            val httpLoggingInterceptor = HttpLoggingInterceptor()
            httpLoggingInterceptor.level = loggingLevel
            return httpLoggingInterceptor
        }
    }

}
